src/Xamarin.Android.Build.Tasks/Linker/MonoDroid.Tuner/FixAbstractMethodsStep.cs

```diff
@@ -22,9 +22,13 @@ protected override void ProcessAssembly (AssemblyDefinition assembly)
 			if (!Annotations.HasAction (assembly))
 				Annotations.SetAction (assembly, AssemblyAction.Skip);
 
-			bool changed = FixAbstractMethods (assembly);
 
-			if (changed) {
+			if (IsProductOrSdkAssembly (assembly))
+				return;
+
+			CheckAppDomainUsageUnconditional (assembly, (string msg) => Context.Logger.LogMessage (MessageImportance.High, msg));
```

Comment:
This needs a warning code, and then we usually write a small doc to put in: https://github.com/xamarin/xamarin-android/tree/master/Documentation/guides/messages

Reply:
Whoops the warning is actually this line: https://github.com/xamarin/xamarin-android/pull/3974/files#diff-fa265e57634387510f2adaed81eb2fd3R58